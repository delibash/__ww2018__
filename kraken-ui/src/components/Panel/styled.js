import styled from 'styled-components';
// import { colors } from './../../constants';

const StyledHeader = styled.h2`
  margin: 1rem 1.8rem 1.8rem;
`;

const StyledContent = styled.section`
  margin: 0 1.8rem;
`;

export {
  StyledHeader,
  StyledContent
};