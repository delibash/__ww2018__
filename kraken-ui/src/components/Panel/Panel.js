import React from 'react';
import PropTypes from 'prop-types';
import { StyledHeader, StyledContent } from './styled';

const Panel = ({children, ...props}) => {
  return (
    <>
      <StyledHeader>{props.header}</StyledHeader>
      <StyledContent>{children}</StyledContent>
    </>
  );
};

Panel.propTypes = {
  children: PropTypes.element.isRequired,
  header: PropTypes.string.isRequired
};

export default Panel;