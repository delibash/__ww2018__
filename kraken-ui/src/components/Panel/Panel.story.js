import React from 'react';
import { storiesOf } from '@storybook/react';
import Panel from './Panel';

storiesOf('Panel', module)
  .add('Default Panel', () => (
    <Panel header="Panel Header">
      <section>Panel Conent</section>
    </Panel>
  ));
