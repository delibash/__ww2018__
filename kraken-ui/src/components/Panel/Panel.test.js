import React from 'react';
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import Panel from './Panel';

test('Panel', () => {
  const tree = renderer.create(<Panel header="Some Header">{<section />}</Panel>).toJSON();
  expect(tree).toMatchSnapshot();
  // expect(tree).toHaveStyleRule('cursor', 'pointer');
});
