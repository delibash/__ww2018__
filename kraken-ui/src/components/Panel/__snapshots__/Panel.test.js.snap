// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel 1`] = `
Array [
  .c0 {
  margin: 1rem 1.8rem 1.8rem;
}

<h2
    className="c0"
  >
    Some Header
  </h2>,
  .c0 {
  margin: 0 1.8rem;
}

<section
    className="c0"
  >
    <section />
  </section>,
]
`;
